### DISCUSSION TITLE
v3.0.0-alpha

### DISCUSSION THREAD

**Discussion by VishnuSanal** [VishnuSanal/Quotes#113] [Announcements]
## v3.0.0-alpha -> What's Changed

- Background images can now be picked from Unsplash
- New implementation of Fonts

Full Changelog -> https://github.com/VishnuSanal/Quotes/compare/v2.10.1...v3.0.0-alpha

<hr /><em>This discussion was created from the release <a href='https://github.com/VishnuSanal/Quotes/releases/tag/v3.0.0-alpha'>v3.0.0-alpha</a>.</em>

**Comment by VishnuSanal** [comment]
Requesting @IzzySoft to pick [this](https://github.com/VishnuSanal/Quotes/releases/tag/v3.0.0-alpha) pre-release up for IzzyOnDroid. Thanks!

**Comment by IzzySoft** [comment]
I hear and obey – but have 2 questions in return:

* shall I pick up all pre-releases from now on, or just this single one? My updater is currently set to skip pre-releases.
* mind to fix formatting of the `full_description.txt`? Description looks [quite wild now](https://apt.izzysoft.de/fdroid/index/apk/phone.vishnu.quotes). As you indented all bullet points, the lists are either seen as code blocks (when interpreted as Markdown) – or the entire description results in a single big wall of text (when taken as-is). Just performing NL2BR will look ugly as well… The following would render nicely when taken as Markdown (in my repo), and look OK with NL2BR (F-Droid):

```markdown
<b>Quotes Status Creator</b>

Quotes Status Creator lets you share quotations as status images on social media – Quotes Status Creator App

* Lots of inspirational quotes that can be shared on social media as a high definition images
* Quotes Status Creation without any Image Editing whatsoever — in the same user-friendly UI
* Browse through Best Quotes → Get Inspired → Share as Status and Spread Positivity

<b>⚙How it Works</b>

Quotes Status Creator App presents you with Famous Quotations. Browse through them, get inspired, share status via social media and spread positivity!

<b>🚀Features</b>

* <b>🔖Favorite Quotes</b> — Add quotes to favorites
* <b>🔀Share Quotes</b> — Share quotes as a high definition images on social media
* <b>🖼️Custom Backgrounds</b> — Personalized quote statuses with custom background images, solid background colors or images from gallery
* <b>🎨Custom Card Colors</b> — Personalized quote statuses with custom card colors
* <b>🔠Custom Fonts</b> — Personalize quotes statuses with custom fonts
```

**Reply by IzzySoft** [reply]
PS: You can also use `<a href` in your description. And btw I wonder why a FOSS app advertises an Instagram channel and not a Pixelfed one :wink:

I've locally applied the above mentioned changes, so after tomorrow's sync you'll see the difference.

**Reply by VishnuSanal** [reply]
@IzzySoft Ping

> * shall I pick up all pre-releases from now on, or just this single one? My updater is currently set to skip pre-releases.

Sure, just clarifying that release versions & pre-release versions will be different from each other (similar to F-Droid's main repo - (screenshot from [AppManager](https://f-droid.org/en/packages/io.github.muntashirakon.AppManager))) and won't be merged with the release versions :)

<img src="https://user-images.githubusercontent.com/50027064/164079551-0beb9548-e01b-41d4-8626-4742590b1593.jpg" width=400/>


> * mind to fix formatting of the `full_description.txt`? Description looks [quite wild now](https://apt.izzysoft.de/fdroid/index/apk/phone.vishnu.quotes). As you indented all bullet points, the lists are either seen as code blocks (when interpreted as Markdown) – or the entire description results in a single big wall of text (when taken as-is). Just performing NL2BR will look ugly as well… The following would render nicely when taken as Markdown (in my repo), and look OK with NL2BR (F-Droid):
> 
> ```gfm
> <b>Quotes Status Creator</b>
> 
> Quotes Status Creator lets you share quotations as status images on social media – Quotes Status Creator App
> 
> * Lots of inspirational quotes that can be shared on social media as a high definition images
> * Quotes Status Creation without any Image Editing whatsoever — in the same user-friendly UI
> * Browse through Best Quotes → Get Inspired → Share as Status and Spread Positivity
> 
> <b>⚙How it Works</b>
> 
> Quotes Status Creator App presents you with Famous Quotations. Browse through them, get inspired, share status via social media and spread positivity!
> 
> <b>🚀Features</b>
> 
> * <b>🔖Favorite Quotes</b> — Add quotes to favorites
> * <b>🔀Share Quotes</b> — Share quotes as a high definition images on social media
> * <b>🖼️Custom Backgrounds</b> — Personalized quote statuses with custom background images, solid background colors or images from gallery
> * <b>🎨Custom Card Colors</b> — Personalized quote statuses with custom card colors
> * <b>🔠Custom Fonts</b> — Personalize quotes statuses with custom fonts
> ```

Sure! I will update my file too :)

> PS: You can also use `<a href` in your description.

TBH, those were copied from Play Store's and never bothered to convert 'em to markdown 🙈 

> And btw I wonder why a FOSS app advertises an Instagram channel and not a Pixelfed one :wink:

XD! Should look into it!

> I've locally applied the above mentioned changes, so after tomorrow's sync you'll see the difference.

Thanks, can you please give me an approx time (and time-zone too!)?

Thanks again! :D

**Reply by IzzySoft** [reply]
> just clarifying that release versions & pre-release versions will be different from each other

The difference your screenshot shows is not provided by my repo (the "suggested version" here is always the latest version, as `CurrentVersion` and `CurrentVersionCode`, which result in the "suggested version", are not used here). So I wonder how this screenshot came to pass: is your app in another repo as well, e.g. at F-Droid? I couldn't find it there.

Which is why I wonder that you suddenly want the alphas picked up as well. I fully understand that if the stable versions are available at F-Droid, but that's not the case here. But if you confirm this is still what you want (even without the distinction just mentioned), I can of course change the config here to pick them up regularly – it's as simple as a 0/1 toggle.

> TBH, those were copied from Play Store's and never bothered to convert 'em to markdown

Well… F-Droid.org does not support markdown, just my repo does. That's why what I recommend is something I call "Markdown Lite", [see here for details](https://gitlab.com/IzzyOnDroid/repo/-/wikis/Fastlane) – this works well for all 3 places (F-Droid, my repo & Goggle) AFAIK. Thanks for updating then! It's mainly the indention – and the type of bullets used. While on it you might think about if the mention of "open source" plus Github links are needed – at least with my repo (and F-Droid's, should your app go there) have those links next to the app description. But sure, if you want to use Fastlane to deploy to Play Store as well, you might wish to keep that.

> XD! Should look into it!

Yay :star_struck:

> Thanks, can you please give me an approx time (and time-zone too!)?

Around 6 pm UTC (summer; 7 pm UTC in winter – fck the constant changing of time settings requiring to specify two of them…)

> Thanks again! :D

You're welcome!

**Reply by shuvashish76** [reply]
> is your app in another repo as well, e.g. at F-Droid...

Not yet. Our final goal is to remove [non-free libraries](https://github.com/VishnuSanal/Quotes/discussions/70#discussioncomment-2159231) & publish on F-Droid.

> you suddenly want the alphas picked up as well...

Because this is not a daily use category app (e.g. filemanager, browser etc..) & that's why the community is very small. We want you to pick up alphas as well for more user feedback. Again the goal is to publish final [v3](https://github.com/VishnuSanal/Quotes/milestone/7) stable version at F-Droid repo.

CMIIW @VishnuSanal

**Reply by VishnuSanal** [reply]
@IzzySoft Ping

> So I wonder how this screenshot came to pass: is your app in another repo as well, e.g. at F-Droid? I couldn't find it there.

It's not a screenshot of my app, it is the screenshot of [AppManager](https://f-droid.org/en/packages/io.github.muntashirakon.AppManager) which is available on F-Droid.

> Which is why I wonder that you suddenly want the alphas picked up as well. I fully understand that if the stable versions are available at F-Droid, but that's not the case here. But if you confirm this is still what you want (even without the distinction just mentioned), I can of course change the config here to pick them up regularly – it's as simple as a 0/1 toggle.

Agree with @shuvashish76, please configure the bot to pick 'em up! The user can anyways see `alpha` in the version name, so it'd be good, I feel.

> > is your app in another repo as well, e.g. at F-Droid...
> 
> Not yet. Our final goal is to remove [non-free libraries](https://github.com/VishnuSanal/Quotes/discussions/70#discussioncomment-2159231) & publish on F-Droid.

Yes. Also, hoping to see no `anti-features` when IzzyOnDroid picks v3.0.0-alpha up 🥳

Thanks! :D

**Reply by shuvashish76** [reply]
> hoping to see no anti-features when IzzyOnDroid picks v3.0.0-alpha up

Unspash API 🙈 #53

```
Anti-features
- Promotes non-free network services
```

@IzzySoft please do post your scanner result here & let us know if we're currently eligible for F-Droid.

**Reply by IzzySoft** [reply]
> please configure the bot to pick 'em up!

Done!

> The user can anyways see `alpha` in the version name

Yes, all can SEE it. Some before they update (manually), others when the update has been applied already (auto-update via privileged extension) :wink:

> Yes. Also, hoping to see no `anti-features` when IzzyOnDroid picks v3.0.0-alpha up

Oh – they probably get removed with a delay, when my checker runs again this evening. But now you mention it I've checked it immediately and can confirm their removal – so with today's sync the AntiFeatures will be gone. Congrats! :partying_face: :tada: Ready for F-Droid then :smiley:

Speaking of which: options once your app hits the F-Droid catalog:

* it gets removed from my repo after a reasonable overlap (usually ~10 days; this is the default)
* keeping it in my repo including "pre-releases"
* keeping it in my repo for pre-releases only (having F-Droid shipping the stable releases)

The last item is achieved more easily by using dedicated tag names, e.g. `fdroid-<versionCode>` for stable and `pre-<versionCode>` for the others. W would then check the tag names against the patterns, and each would only pick up theirs. You might already wish to think about this so you can decide when the day comes :wink:

**Comment by IzzySoft** [comment]
As you asked for a full report, here you go:

```
Libraries detected:
-------------------
* Android Support v4 (/android/support/v4): Development Framework, Apache-2.0
* AndroidX Activity (/androidx/activity): Utility, Apache-2.0
* Android Jetpack Annotations (/androidx/annotation): Utility, Apache-2.0
* AppCompat (/androidx/appcompat): Utility, Apache-2.0
* Browser (/androidx/browser): Utility, Apache-2.0
* Constraint Layout Library (/androidx/constraintlayout): Utility, Apache-2.0
* Androidx Core (/androidx/core): Utility, Apache-2.0
* Android Emoji2 Compat (/androidx/emoji2): UI Component, Apache-2.0
* Androidx Fragment (/androidx/fragment/app): Development Aid, Apache-2.0
* Lifecycle (/androidx/lifecycle): Utility, Apache-2.0
* Media (/androidx/media): Utility, Apache-2.0
* Room (/androidx/room): Utility, Apache-2.0
* Recyclerview (/androidx/recyclerview): Utility, Apache-2.0
* Android Activity Saved State (/androidx/savedstate): Utility, Apache-2.0
* Android Jetpack VersionedParcelable (/androidx/versionedparcelable): Utility, Apache-2.0
* AndroidX Widget ViewPager2 (/androidx/viewpager2): UI Component, Apache-2.0
* Google Material Design (/com/google/android/material): Utility, Apache-2.0
* Dexter (/com/karumi/dexter): Utility, Apache-2.0
* Android Unsplash (Unofficial) (/com/keenencharles/unsplash): Utility, MIT
* Slide To Act (/com/ncorti/slidetoact): UI Component, MIT
* ColorPickerView (/com/skydoves/colorpickerview): UI Component, Apache-2.0
* OkHttp (/com/squareup/okhttp): Utility, Apache-2.0
* Picasso (/com/squareup/picasso): Utility, Apache-2.0
* uCrop (/com/yalantis/ucrop): Utility, Apache-2.0
* Kotlin (/kotlin): Utility, Apache-2.0
* kotlinx.coroutines (/kotlinx/coroutines): Utility, Apache-2.0
* ACRA (/org/acra): Utility, Apache-2.0

No offending libs found.
```

So I see no stoppers **in the resulting APK** – F-Droid also scans the "input", so we can talk about that only once a scan was performed there.

**Comment by IzzySoft** [comment]
> please configure the bot to pick 'em up!

Shall I still pick up pre-releases now the latest ones are stable again?

**Reply by VishnuSanal** [reply]
Yep, you can ignore the pre release versions for now, thanks. :) Until, we are live on FDroid main repo (#101), the RFP is blocked for some reason, we can remove it from IzzyOnDroid then, freeing your resources. :D

**Reply by IzzySoft** [reply]
Thanks! Skipping pre-releases from now on.

> freeing your resources

Not that the app consumes much, the APK is still < 3 MB.

Ugh… while checking that I see there's a problem with your `full_description.txt` in Fastlane, resulting in this:

![image](https://github.com/VishnuSanal/Quotes/assets/6781438/16ab4300-87f4-4527-92d4-93b735b89d11)

Look for yourself to get the [full picture](https://apt.izzysoft.de/fdroid/index/apk/phone.vishnu.quotes).

Could you please use plain asterisks (`*`) as bullet-points, and have them **NOT** indented? 4+ leading spaces mark a line as "code block" (Markdown specification).

**Reply by VishnuSanal** [reply]
✅ done, should I create a release, or can you pull it?

**Reply by IzzySoft** [reply]
```
$ iod repo fastlane phone.vishnu.quotes
phone.vishnu.quotes: check if repo contains FUNDING.yml
phone.vishnu.quotes: looking for 'https://api.github.com/repos/VishnuSanal/Quotes/contents/.github'
phone.vishnu.quotes: FUNDING.yml already up-to-date.
phone.vishnu.quotes: checking Fastlane for per-release changelogs
phone.vishnu.quotes: cleaning up older changelogs
phone.vishnu.quotes: fetched 'https://github.com/VishnuSanal/Quotes/raw/master/fastlane/metadata/android/en-US/changelogs/63.txt'
phone.vishnu.quotes: calling 'getFastlaneMeta(github,[host:github.com,owner:VishnuSanal,repo:Quotes,path:/fastlane/metadata/android])'
phone.vishnu.quotes: FastlaneFeatures title,shortdesc,fulldescMD,icon,featureGraphic,screenshotsJPG
phone.vishnu.quotes: looking for 'https://api.github.com/repos/VishnuSanal/Quotes/contents/fastlane%2Fmetadata%2Fandroid'
phone.vishnu.quotes: looking for 'https://api.github.com/repos/VishnuSanal/Quotes/contents/fastlane%2Fmetadata%2Fandroid%2Fen-US'
phone.vishnu.quotes: looking for 'https://api.github.com/repos/VishnuSanal/Quotes/contents/fastlane%2Fmetadata%2Fandroid%2Fen-US%2Fchangelogs'
phone.vishnu.quotes: looking for 'https://api.github.com/repos/VishnuSanal/Quotes/contents/fastlane%2Fmetadata%2Fandroid%2Fen-US%2Fimages'
phone.vishnu.quotes: looking for 'https://api.github.com/repos/VishnuSanal/Quotes/contents/fastlane%2Fmetadata%2Fandroid%2Fen-US%2Fimages%2FphoneScreenshots'
phone.vishnu.quotes: checking locale 'en-US'
phone.vishnu.quotes: replacing 'metadata/phone.vishnu.quotes/en-US/full_description.txt' with fromMD
phone.vishnu.quotes: updating 'repo/phone.vishnu.quotes/en-US/phoneScreenshots/1.png' as JPG
phone.vishnu.quotes: updating 'repo/phone.vishnu.quotes/en-US/phoneScreenshots/2.png' as JPG
phone.vishnu.quotes: updating 'repo/phone.vishnu.quotes/en-US/phoneScreenshots/3.png' as JPG
phone.vishnu.quotes: updating 'repo/phone.vishnu.quotes/en-US/phoneScreenshots/4.png' as JPG
phone.vishnu.quotes: updating 'repo/phone.vishnu.quotes/en-US/phoneScreenshots/5.png' as JPG
phone.vishnu.quotes: updating 'repo/phone.vishnu.quotes/en-US/phoneScreenshots/6.png' as JPG
phone.vishnu.quotes: updating 'repo/phone.vishnu.quotes/en-US/phoneScreenshots/7.png' as JPG
phone.vishnu.quotes: cross-checking for obsolete screenshots
phone.vishnu.quotes: screenshots in Fastlane: 1,2,3,4,5,6,7
phone.vishnu.quotes: local screenshots checked: 1,2,3,4,5,6,7
```

![image](https://github.com/VishnuSanal/Quotes/assets/6781438/e333ad35-48f1-4f81-9224-e2007e073686)

Thanks! Will go public with the next sync around 7 pm UTC.
